### DISCUSSION TITLE
Using Google Coral on Synology Docker without starting from CLI

### DISCUSSION THREAD

**Discussion by wills106** [blakeblackshear/frigate#364] [Show and tell]
I have found if you start Frigate on a Synology NAS you don't get USB Coral support due to the fact you can't set `/dev/bus/usb:/dev/bus/usb` in the GUI.
You have to SSH into your Synology and manually start from the command line.

If you don't feel comfortable doing so I have created the json settings for you. If you paste the code below the image into a file called frigate.syno.json you can then import that file and start a container with `/dev/bus/usb:/dev/bus/usb` working.

You will need to manually pull the Docker Image though first, but you don't need to create a container from it.
Replace `"image" : "blakeblackshear/frigate:0.8.0-beta1-amd64"` with the version you pull down.
Also change your directories to suit.

![image](https://user-images.githubusercontent.com/18155231/101911667-9719be00-3bb8-11eb-91f1-77963fa182f5.png)

```
{
   "cap_add" : [],
   "cap_drop" : [],
   "cmd" : "/run.sh",
   "cpu_priority" : 50,
   "devices" : [],
   "enable_publish_all_ports" : false,
   "enable_restart_policy" : false,
   "enabled" : true,
   "env_variables" : [
      {
         "key" : "PATH",
         "value" : "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      },
      {
         "key" : "FLASK_ENV",
         "value" : "development"
      },
      {
         "key" : "DEBIAN_FRONTEND",
         "value" : "noninteractive"
      },
      {
         "key" : "LIBVA_DRIVER_NAME",
         "value" : "i965"
      }
   ],
   "exporting" : false,
   "image" : "blakeblackshear/frigate:0.8.0-beta1-amd64",
   "is_ddsm" : false,
   "is_package" : false,
   "links" : [],
   "memory_limit" : 0,
   "name" : "frigate",
   "network" : [
      {
         "driver" : "bridge",
         "name" : "bridge"
      }
   ],
   "network_mode" : "default",
   "port_bindings" : [
      {
         "container_port" : 1935,
         "host_port" : 1935,
         "type" : "tcp"
      },
      {
         "container_port" : 5000,
         "host_port" : 5000,
         "type" : "tcp"
      }
   ],
   "privileged" : true,
   "shortcut" : {
      "enable_shortcut" : false
   },
   "use_host_network" : false,
   "volume_bindings" : [
      {
         "host_volume_file" : "/docker/hass.io/media/frigate/clips",
         "mount_point" : "/media/frigate/clips",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/docker/hass.io/media/frigate/recordings",
         "mount_point" : "/media/frigate/recordings",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/docker/frigate",
         "mount_point" : "/config",
         "type" : "ro"
      },
      {
         "host_absolute_path" : "/dev/bus/usb",
         "mount_point" : "/dev/bus/usb",
         "type" : "rw"
      },
      {
         "host_absolute_path" : "/etc/localtime",
         "mount_point" : "/etc/localtime",
         "type" : "ro"
      }
   ]
}
```

**Comment by nunofgs** [comment]
This approach doesn’t seem to work on DSM7. I’m getting the following error:

```
detector.coral                 INFO    : Starting detection process: 35
frigate_1          | frigate.edgetpu                INFO    : Attempting to load TPU as usb
frigate_1          | frigate.edgetpu                INFO    : No EdgeTPU detected.
frigate_1          | Process detector:coral:
frigate_1          | Traceback (most recent call last):
frigate_1          |   File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 152, in load_delegate
frigate_1          |     delegate = Delegate(library, options)
frigate_1          |   File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 111, in __init__
frigate_1          |     raise ValueError(capture.message)
frigate_1          | ValueError
frigate_1          | 
frigate_1          | During handling of the above exception, another exception occurred:
frigate_1          | 
frigate_1          | Traceback (most recent call last):
frigate_1          |   File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
frigate_1          |     self.run()
frigate_1          |   File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
frigate_1          |     self._target(*self._args, **self._kwargs)
frigate_1          |   File "/opt/frigate/frigate/edgetpu.py", line 124, in run_detector
frigate_1          |     object_detector = LocalObjectDetector(tf_device=tf_device, num_threads=num_threads)
frigate_1          |   File "/opt/frigate/frigate/edgetpu.py", line 63, in __init__
frigate_1          |     edge_tpu_delegate = load_delegate('libedgetpu.so.1.0', device_config)
frigate_1          |   File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 154, in load_delegate
frigate_1          |     raise ValueError('Failed to load delegate from {}\n{}'.format(
frigate_1          | ValueError: Failed to load delegate from libedgetpu.so.1.0
```


Anyone have any luck with DSM7?

**Reply by tnagels** [reply]
Did you get a solution for this? Or is this one of the many devices which are no longer supported by DSM7?

**Reply by nunofgs** [reply]
Not really :(

I can see the device show up with lsusb on the host as well as inside a container, but that’s as far as I got.

**Comment by gadget-man** [comment]
I've just got a Coral.ai usb stick and found this incredibly useful. It's worth noting also for others coming across this page that to get Coral.ai working on an Synology NAS DNS 6.xx with Frigate, you don't need to bother with any of the coral.ai setup scripts - you can just plug it in, make the changes proposed above to link to the volume and restart Frigate. 

Looking forward to the gurus finding a way to make it still work on DSM7!

**Reply by aptonline** [reply]
Not sure if this is of use but USB drivers for DSM 7 are now available in test form at http://www.jadahl.com/.

**Reply by IZZE2000** [reply]
Any chance you (or someone else) can describe how to do this step by step on my DS218+ with DNS 6.xx? 
I do have the Docker package installed as well but i do get stuck.

Btw. I'm running HA under VMM, works perfectly well!
Btw2. if you get it to work under DNS 7 i'm all ears as well. 
Cheers

**Comment by zerquanta** [comment]
Has anyone managed to get Coral.ai  USB stick to work on DSM 7?

**Reply by BeardedTinker** [reply]
I have BUT not in Docker - I'm using VMM. Home Assistant in VM with Frigate AddOn is working at this point.

**Reply by federicotravaini** [reply]
Is it working? Mine keeps "crashing" every 10 minutes, even if the logs say EDGEPTU FOUND. It's like the USB CORAL is found, but Frigate can't really use it for detection. Did you do any special setup? I just add the USB in the UI of the VM settings

**Reply by UnknownUser6** [reply]
Did you get it to work stable? I seem to have the same problem. I get to the point "EDGEPTU FOUND", but it crashes every couple minutes.

**Reply by BeardedTinker** [reply]
For me it worked anywhere from a week down to 15.20 minutes. After that it would disconnect and I would need to unplug and plug it again. As I'm using Surveillance station, this was more of a test - works, yes, reliably for long term use, no.

**Comment by jcastro** [comment]
This has worked perfectly for me, thanks a lot!
I also had to change the 5000 Port as that's what's my Syno is using, but other than that it's working ok. Also the HA integration works ok, just need to remember to use the new port you set if you changed from the original 5000 one.

**Comment by aptonline** [comment]
I've yet to get my Coral USB stick but thought I'd still follow this guide to setup Frigate on my Synology but I'm seeing the following in the logs and the Frigate UI just shows the logo and spinner:

```
2021-08-12 14:41:04 | stderr | root                           INFO    : Waiting for detection process to exit gracefully...
-- | -- | --
2021-08-12 14:41:04 | stderr | peewee.sqliteq                 INFO    : writer received shutdown request, exiting.
2021-08-12 14:41:04 | stderr | frigate.stats                  INFO    : Exiting watchdog...
2021-08-12 14:40:24 | stderr | frigate.watchdog               INFO    : Exiting watchdog...
2021-08-12 14:40:24 | stderr | frigate.events                 INFO    : Exiting event cleanup...
2021-08-12 14:40:19 | stderr | frigate.object_processing      INFO    : Exiting object processor...
2021-08-12 14:40:14 | stderr | frigate.record                 INFO    : Exiting recording maintenance...
2021-08-12 14:40:14 | stderr | frigate.events                 INFO    : Exiting event processor...
2021-08-12 14:40:14 | stderr | frigate.app                    INFO    : Stopping...
2021-08-12 14:40:14 | stderr | frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...
2021-08-12 14:39:54 | stderr |  
2021-08-12 14:39:54 | stderr | ValueError: Failed to load delegate from libedgetpu.so.1.0
2021-08-12 14:39:54 | stderr | raise ValueError('Failed to load delegate from {}\n{}'.format(
2021-08-12 14:39:54 | stderr | File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 154, in load_delegate
2021-08-12 14:39:54 | stderr | edge_tpu_delegate = load_delegate('libedgetpu.so.1.0', device_config)
2021-08-12 14:39:54 | stderr | File "/opt/frigate/frigate/edgetpu.py", line 63, in __init__
2021-08-12 14:39:54 | stderr | object_detector = LocalObjectDetector(tf_device=tf_device, num_threads=num_threads)
2021-08-12 14:39:54 | stderr | File "/opt/frigate/frigate/edgetpu.py", line 124, in run_detector
2021-08-12 14:39:54 | stderr | self._target(*self._args, **self._kwargs)
2021-08-12 14:39:54 | stderr | File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
2021-08-12 14:39:54 | stderr | self.run()
2021-08-12 14:39:54 | stderr | File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
2021-08-12 14:39:54 | stderr | Traceback (most recent call last):
2021-08-12 14:39:54 | stderr |  
2021-08-12 14:39:54 | stderr | During handling of the above exception, another exception occurred:
2021-08-12 14:39:54 | stderr |  
2021-08-12 14:39:54 | stderr | ValueError
2021-08-12 14:39:54 | stderr | raise ValueError(capture.message)
2021-08-12 14:39:54 | stderr | File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 111, in __init__
2021-08-12 14:39:54 | stderr | delegate = Delegate(library, options)
2021-08-12 14:39:54 | stderr | File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 152, in load_delegate
2021-08-12 14:39:54 | stderr | Traceback (most recent call last):
2021-08-12 14:39:54 | stderr | Process detector:coral:
2021-08-12 14:39:54 | stderr | frigate.edgetpu                INFO    : No EdgeTPU detected.
2021-08-12 14:39:54 | stderr | frigate.edgetpu                INFO    : Attempting to load TPU as usb
2021-08-12 14:39:54 | stderr | frigate.app                    INFO    : Capture process started for back: 40
2021-08-12 14:39:54 | stderr | frigate.app                    INFO    : Camera processor started for back: 39
2021-08-12 14:39:54 | stderr | detector.coral                 INFO    : Starting detection process: 36
2021-08-12 14:39:53 | stderr | frigate.mqtt                   INFO    : MQTT connected
2021-08-12 14:39:53 | stderr | peewee_migrate                 INFO    : There is nothing to migrate
2021-08-12 14:39:53 | stderr | There is nothing to migrate
2021-08-12 14:39:53 | stderr | peewee_migrate                 INFO    : Starting migrations
2021-08-12 14:39:53 | stderr | Starting migrations
2021-08-12 14:39:42 | stdout | ...done.
2021-08-12 14:39:42 | stdout | * Starting nginx nginx
```

Is this simply because I don't have Coral plugged in at this stage?

**Reply by tnagels** [reply]
Yes. If it says so in the log. I seem to remember that if you don't want to use the TPU, you need to put that in the config explicitly.

**Reply by aptonline** [reply]
Thanks for confirming @tnagels, looking at the config page I can't see where I would exclude the TPU in the config: https://blakeblackshear.github.io/frigate/configuration/index

**Reply by wills106** [reply]
If you have something like:

```
detectors:
  coral:
    type: edgetpu
    device: usb
```

You need to replace it with:

```
detectors:
  cpu1:
    type: cpu
```

https://blakeblackshear.github.io/frigate/configuration/detectors

**Reply by aptonline** [reply]
ah thanks. will give it a try.

**Reply by aptonline** [reply]
working perfectly, thanks @wills106  , @tnagels  👍🏼

**Reply by slepiavka** [reply]
@aptonline I believe this makes use of multi-core CPUs? Not sure, but increased my inference speeds:
```
detectors:
  cpu1:
    type: cpu
  cpu2:
    type: cpu

**Comment by deviant77** [comment]
Has anyone got this to work with the 0.9.0 betas?

It fails to start with both the blakeblackshear/frigate:0.9.0-beta1-amd64 and blakeblackshear/frigate:0.9.0-beta2-amd64 images.
It works with blakeblackshear/frigate:0.8.4-amd64.

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.

ifelse: fatal: unable to exec /run.sh: No such file or directory
[cmd] /run.sh exited 127

[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

**Reply by dqx42** [reply]
For those trying to get an edge TPU running with a synology NAS, I found that I needed a powered USB hub to get it working on my DS916+. The usb ports on the NAS are apparently unable to supply sufficient power for the device and frigate within the docker container was crashing on start up as it tries to initialise the TPU. Once I give it power through the hub, everything starts up fine - running DSM7.

**Reply by nunofgs** [reply]
@dqx42 I just tried your suggestion but wondering if my powered hub is even working properly. I'm still getting `Failed to load delegate from libedgetpu.so.1.0`

Can you share your docker container settings? what volumes/devices are you mounting?

**Reply by dqx42** [reply]
The behaviour I was seeing was the NAS USB dropping the TPU when it tried to boot. I plugged the TPU into USB and saw it appear. Start the docker container, I got an error and the TPU was no longer showing in lsusb. I tried this on both the front an back usb ports on the NAS. As soon as I gave it separate power the problem went away...
The relevant part of my docker compose script below. 
Good luck!

 frigate:
    container_name: frigate
    image: blakeblackshear/frigate:stable-amd64
    shm_size: "1024m"
    privileged: true
    hostname: frigate
    domainname: [myhome.domain.com]
    devices:
      - /dev/bus/usb:/dev/bus/usb
      - /dev/dri:/dev/dri
    ports:
      - 5005:5000
      - 1935:1935
    volumes:
      - /dev/bus/usb:/dev/bus/usb
      - /etc/localtime:/etc/localtime:ro
      - /volume1/frigate/config/frigate.yml:/config/config.yml:ro
      - /volume1/frigate/media:/media/frigate
      - type: tmpfs
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    restart: unless-stopped

**Reply by S-Przybylski** [reply]
I tried today on DS1621+ with DSM7 without luck (docker)! Neither with USB3 Hub (with external power) nor without. Btw: While I compare the USB usage between RASPI4 and DS1621+, I saw at RASPI4 that the device first comes with an Vendor-ID 1a6e (prod 089a) and after initial load of the firmware this changes to Vendor-ID 18d1 (prod 9302).
Any suggestions in regards to DS1621+/other current Syno-Devices?

**Reply by S-Przybylski** [reply]
Next try: 
Installed a virtual machine in DSM7 and added docker/docker-compose: 
1. debian 11 doesn't work - dma errors using Coral stick TPU
2. ubuntu server with frigate docker coantiner and Coral TPU seems to work - currently i see no dma errors

Note: While booting the vm, the usb device need to be changed from 1a6e to 18d1. After some docker container restarts it will run then.

Does anybody have an idea, how to avoid the manual intervention changing the usb devices at start time?

**Reply by S-Przybylski** [reply]
Update: got also dma issues with ubuntu:
 xhci_hcd 0000:00:04.0: ERROR Transfer event TRB DMA ptr not part of current TD ep_index 2 comp_code 1

Seems to be related to DSM7
Best will be relocating Frigate to NUC :-(

**Reply by mlebaugh** [reply]
> The behaviour I was seeing was the NAS USB dropping the TPU when it tried to boot. I plugged the TPU into USB and saw it appear. Start the docker container, I got an error and the TPU was no longer showing in lsusb. I tried this on both the front an back usb ports on the NAS. As soon as I gave it separate power the problem went away... The relevant part of my docker compose script below. Good luck!

powered hub resolved my issue, thanks!

**Comment by federicotravaini** [comment]
Might be a dumb question but: isn't installing Portainer easier? You can get full control of Docker with a nice UI

**Reply by juagicre** [reply]
i'm interested! but how to install it in Synology DSM if it is not available in the offered/official packages???

**Reply by juagicre** [reply]
I did follow a guideline suggested in next line to install it, seems to work fine, now i need to learn how to use it and mount the usb device :) I believe that's all I need to get the coral being detected by the frigate container :P

**Comment by Floriszz** [comment]
Another excellent tip I found on internet with installing and updaing Portainer Docker Container is creating a task under task scheduler. To see an example how this is done for Portainer look for 'Synology: 30 Second Portainer Install Using Task Scheduler & Docker' from Marius Hosting.
![image](https://user-images.githubusercontent.com/19168937/178163090-d8cc84e1-06da-45be-8051-58e501f425d8.png)

**Reply by pShota** [reply]
Hi @Floriszz 
I cant get this to work,
can you see the device inside frigate container?

**Reply by Floriszz** [reply]
Can you elaborate on the 'can't get this to work'.?
So far my problem is that the Docker image of Frigate image keeps rebooting. But that is probably because of incorrect lines in the settings. It is complaining about naming incorrect.
Also I did not have a mqtt broker set up before. Creating a mqtt broker on a raspberry PI with Home Assistant right now, as I received my Coral yesterday!(yeah, waited for it almost a year after order) I started to pick up the work to get it working.
I will be looking into getting the Docker script correct over the next days. I'm still on DSM 6.x(latest). Don't want to upgrade to DSM 7 before getting it to work first.

**Reply by pShota** [reply]
HI @Floriszz , thanks for responsing
my frigate container cannot find coral usb by just using this method,
later I find out another post that someone did an excellent effort to find out the root cause and also provided a solution.
see https://github.com/weltenwort/frigate-synology-dsm7
this contain the fix for the usb driver of synology, I use it without problem now with my DS620Slim, cheers

**Reply by IZZE2000** [reply]
> HI @Floriszz , thanks for responsing my frigate container cannot find coral usb by just using this method, later I find out another post that someone did an excellent effort to find out the root cause and also provided a solution. see https://github.com/weltenwort/frigate-synology-dsm7 this contain the fix for the usb driver of synology, I use it without problem now with my DS620Slim, cheers

This looks very promising! How long time have you tried it? I'm still on dms6 and had been hesitant to update since I thought the Coral worked even worse on dms7. However if it works fine I'm all in on jumping on the dms7 band wagon :-)
Cheers

**Reply by kaytwo** [reply]
switching from mapping the tpu in via `devices` to `volumes` fixed this error in DSM 6 for me.

**Reply by Floriszz** [reply]
@kaytwo , thank you for the last hurdle. Finally have it working without using SSH or any command line tools or docker compose. 
Using solely the Synology DSM Task Scheduler. (I'm till on DSM 6.x!) As shown in the screenshot in the above post. 
Place the commands below in the task scheduler 'Run command' box, Save and Run the task. 
The exact commands I use now are: (I use Volume2, because I have my docker images on a separate volume in DSM)

docker run -d \
--name frigate \
--restart=unless-stopped \
--mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \
--device /dev/dri/renderD128 \
--shm-size=64m \
--privileged \
-v /volume2/docker/frigate/media:/media/frigate \
-v /volume2/docker/frigate/config/config.yml:/config/config.yml:ro \
-v /etc/localtime:/etc/localtime:ro \
-v /dev/bus/usb:/dev/bus/usb \
-v /volume2/docker/frigate/db:/db \
-e FRIGATE_RTSP_PASSWORD=MyTestCamPassword \
-p 5055:5000 \
-p 1935:1935 \
blakeblackshear/frigate:stable-amd64

My config.yml file is:
mqtt:
  host: 192.168.1.28
  port: 1883
  user: homeassistant
  password: ziesdf8sd89a7fds87fdsa8s7f8sa7fsafd798df7af7dsf7asd98f7sdfweo3
  stats_interval: 60
detectors:
  coral:
    type: edgetpu
    device: usb
database:
  path: /db/frigate.db
birdseye:
  enabled: True
  mode: objects
cameras:
  camera_1: # Backdoor
    ffmpeg:
      inputs:
        - path: rtsp://admin:MyTestCamPassword@192.168.1.40:554/live/av0? # H264 1280x720
          roles:
            - detect
            - rtmp
            - record
    record: # <----- # Enable recording
      enabled: False
    snapshots: # <----- Enable snapshots
      enabled: True
    rtmp:
      enabled: False # <-- RTMP should be disabled if your stream is not H264
    detect:
      enabled: True 
      width: 1280 # <---- update for your camera's resolution
      height: 720 # <---- update for your camera's resolution

![image](https://user-images.githubusercontent.com/19168937/216767381-19a4465a-86b5-4ded-98cc-937e45360152.png)

**Reply by neuhausf** [reply]
@NickM-27 Any chances, that the changes from https://github.com/weltenwort/frigate-synology-dsm7 will be merged into this official repo? I would really like to use USB coral with Synology DSM 7.1

**Reply by kaytwo** [reply]
The ability to mount a coral in the Docker container is merged into Frigate .12, working without problems for me on DSM 7 with the current beta.

**Reply by neuhausf** [reply]
Thats great news (which I obviously missed)! So I'm going to order a stick 😃

**Reply by Floriszz** [reply]
@kaytwo , I already posted the same question at: https://github.com/weltenwort/frigate-synology-dsm7/discussions/94#discussioncomment-5381078
Can you confirm that it will work the way I install/run/update Frigate(through DSM Taks Scheduler) with the commands above?
The trick is that I don't use any cli/SSH to add stuff to DSM software in order for the USB dongle to be available in the docker image. In other words, will the Coral USB dongle be available for the Frigate Container image without the need of adding/changing code in DSM 7.x itself?

**Reply by kaytwo** [reply]
I'm sorry I can't help you on this, I only use `docker-compose` via ssh.

**Reply by neuhausf** [reply]
I used portainer on a synology 918+ to setup a frigate-stack. Works with coral usb and without modifying anything on the synology.
With the coming dsm 7.2 you can (probably) add stacks via UI too. But its very limited.

**Reply by Floriszz** [reply]
Thanks a Million. That is great news! I will update my DSM as well then! Fingers crossed, it will work.

**Reply by dasb00ter** [reply]
@neuhasf could you maybe post your docker compose stack. I am trying to get this to run in VM but it seems that it is the same problem problem with the global chip to google conversion. I think I a may have to run frigate in docker by itself since I cant pass through the usb

**Reply by neuhausf** [reply]
@dasb00ter sure thing:
I'm using a 918+ with DSM 7.1.1. The coral USB is in the USB port in the back.

1. go to your portainer, stacks, editor, paste the following:
2. note: I used port 5000 which is also the default for DSM. Rewrite your DSM port to something other before starting the stack. Otherwise you will get something like "addr/port in use". The reason why I use the default 5000 for frigate: I think there is some hardcoded code in the home-assistant card which points to 5000. But if you don't use home-assistant, you can remap this.
3. note 2: I use go2rtc also as a proxy for scrypted. You may not expose/remap all of this ports.
```
version: "3.9"
services:
  frigate:
    container_name: frigate
    privileged: true
    restart: unless-stopped
    image: ghcr.io/blakeblackshear/frigate:0.12.0-beta10 # use the beta!
    shm_size: "128mb" # update for your cameras
    devices:
      - /dev/dri/renderD128 # for intel hwaccel, needs to be updated for your hardware
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /volume1/docker/configs/frigate/config.yml:/config/config.yml
      - /volume1/docker/configs/frigate/storage:/media/frigate
    ports:
      - "5000:5000"
      - "1984:1984" # go2rtc api
      - "8554:8554" # RTSP via go2rtc
      - "8555:8555" # webrtc via go2rtc
    environment:
      FRIGATE_RTSP_PASSWORD: "12345" # change it
      LIBVA_DRIVER_NAME: i965 # for GPU support - may not be necessary
```

**Reply by dasb00ter** [reply]
@neuhausf thanks for that. The whole endeavor was because I have unsuccessfully been able to pass through a usb google coral to a VM. What is the devices section for if you dont mind me asking. In this case you have /dev/dri/renderD128 is this for decoding the streams with compatible intel processor/gpu

I noticed that you have NOT defined a TPU like in the installation page like : 
```
 devices:
      - /dev/bus/usb:/dev/bus/usb # passes the USB Coral, needs to be modified for other versions
      - /dev/apex_0:/dev/apex_0 # passes a PCIe Coral, follow driver instructions here https://coral.ai/docs/m2/get-started/#2a-on-linux
      - /dev/dri/renderD128 # for intel hwaccel, needs to be updated for your hardware
```
Is it not necessary? I am using a usb google coral which 
I do use home assistant so I followed your advice on the DSM port. Actually I run hassos and this is my first "addon" run outside of the HA hypervisor. Its off topic but if you could point me in the direction of bringing it in to hassos than I would appreciated it

**Reply by neuhausf** [reply]
@dasb00ter About the usb-device. You are correct, but it seems to work without it: 
<img width="453" alt="image" src="https://user-images.githubusercontent.com/7963239/229425707-fc05b299-13d9-40b5-b961-8f501097a9c8.png">
Probably `privileged: true` does the trick. Anyway: Adding the device explicitely does not change anything but doesn't seem to hurt (just tested it).
As for home-assistant: I use HA docker with the frigate-hass-card. Important thing to note: Use the 5.0.0 release. Either the beta, which has just been released or a daily build from the github-actions (i.e. from here, adapt to the latest commit: https://github.com/dermotduffy/frigate-hass-card/actions/runs/4589791919). Just point your card to your frigate installation (port 5000) and use go2rtc for best latency.
I would make sure, that frigate itself is running fine and showing a camera image in the web interface before starting with HA.

**Reply by dasb00ter** [reply]
I was successful following your directions and had frigate running in docker with the google coral. My testing also seems to indicate it was not necessary to add usb under devices. In the frigate web ui I could see my camera with the google coral tpu. 
Buoyed by this success I attempted to use the addon, the .12beta version this time in HACS under HASSOS(home assistant hypervisor) but this did not work. I am not sure what magic the docker version is doing to pull in the google coral but I cant get it done via Virtual machine running hassos and the beta addon.

 Begs the question to switch to an all dockerized version of Home assistant. Thanks about the lovelace card, will look into it. I believe I had pulled it into HA with the Integration. Thanks about the go2rtc tip I have looked at that but embarrassingly I dont understand that concept or how it fits into the streaming puzzle. 

In any case your instruction was essential to my success. I did a synology "oopsy" during all the testing (dont ask :(  sigghhhhh) So now I am rebuilding my array another 7.5 hours to wait

**Reply by chjud** [reply]
Did you have any success with the HASSOS VM? I run the same setup as you on DSM7. The powered USB Hub didn't do the trick. 
The issue clearly starts with Synology not recognizing the TPU, not with HASSOS.
Did anyone find a way to manually/SSH add the TPU so it is recognized as "Google TPU"?

**Reply by dasb00ter** [reply]
I have frigate running in docker and use the hassos proxy to bring it into home assistant. It's working fine and the only thing to remember is some configuration has to be done directly from the webui of the container

**Reply by chjud** [reply]
Thank you for the information, it does seem that Frigate in Docker is the only way to go
